### PR TITLE
Migrate project to Prettier 3.0.0-alpha.4, run tests against Prettier v1 and v2

### LINKED ISSUE #30: Migrate to Prettier v3
Hi!

The Prettier team is currently working on v3(still alpha).
v3 will include breaking changes to the plugin interface.
Please see https://github.com/prettier/prettier/releases/tag/3.0.0-alpha.2 for more information.

I created this issue to help major plugins migrate to Prettier v3.

Myself and @fisker have created a migration guide which you can see here:

https://github.com/prettier/prettier/wiki/How-to-migrate-my-plugin-to-support-Prettier-v3%3F

If you have any questions, feel free to mention us. We can help you.

If you have already migrated to Prettier v3, sorry but please close this issue.

### PR DESCRIPTION
**Pull request by kachkaev**
Closes #30

### CODE REVIEW

File: src/cache.test.ts [hunk 1]
```diff
@@ -74,4 +74,4 @@ test(`correctly deals with cache`, async () => {
   ).toBeGreaterThanOrEqual(
     numberOfFormatCallsInFirstRun + 1 /* for "" */ + 1 /* for GC-d block */,
   );
-}, 10_000);
+}, 15_000);
```
Comment (kachkaev) [inline src/cache.test.ts:77]: Useful on slow CI runners (which GitHub provides time and again)

Example: https://github.com/gicentre/prettier-plugin-elm/actions/runs/3400687572/jobs/5655277602

File: prettier.config.js [hunk 2]
```diff
@@ -1,5 +0,0 @@
-module.exports = {
-  arrowParens: "always",
-  endOfLine: "lf",
-  trailingComma: "all",
-};
```
Comment (kachkaev) [inline prettier.config.js:5]: No longer needed because all used options are now the defaults.

**Comment by fisker** [comment]
Why don't you do

```sh
yarn upgrade prettier@1
yarn test
```

https://github.com/prettier/angular-estree-parser/blob/09541d0d2ae5d99f17477cae597fa6b743a4a415/.github/workflows/ci.yml#L39

**Comment by kachkaev** [comment]
👋 @fisker! I was thinking of doing something like that in CI, but then decided to include three local Prettier versions. This way I can switch between them on the fly while keeping `git status` clean. I might switch to `yarn upgrade` in CI at some point, but I don’t see any problems with the current approach at the moment.

Thanks for taking a look at this PR, let me know if you have any other thoughts!

**Pull request merged**